_config.yml

```diff
@@ -590,11 +590,32 @@ quicklink:
 # Copyright 2015-2018 GrowingIO, Inc. More info available at https://www.growingio.com
 #growingio_analytics: #your projectId
 
+# Global settings for comments system.
+# You can set one type to enable.
+# And need to provide additional configurations are required for those types.
+# See doc: https://theme-next.org/docs/third-party-services/comments-and-widgets/
+# Example:
+#   comments:
+#     type: Disqus # enable disqus.
+#     count: 
+#       page: true
+#       post: true
+# Additional configurations:
+#   disqus:
+#     shortname: shartname
+#     lazyload: false
+comments:
+  # Type list:
+  # Disqus | Disqusjs | Changyan | Valine | LiveRe | Gitment | Gitalk | facebook_comments_plugin | vkontakte 
```

Comment:
From the point of current vies it is good changes. But my future vision somewhat different: I think each user and blogger will be glad if he can use multi-comments option. E.g. 3-5 comment systems at once, which can be switched by tabs (or somehow else). But with this remake this will be impossible, isn't it?

Reply:
As far as I know there is no mainstream blogging system that uses multiple commenting systems. Maybe you can use the questionnaire to determine the user's needs? However, I am not optimistic about it, in most situations one comment system is enough.

Reply:
> questionnaire

In past, when NexT was in Vi's repo, I remember peoples ask for this feature. And in many cases I want at least to comments system for provide freedom of choice to users. I saw it many times where blogs on WP have multi-comments and this options looks great. Anyway, possibility for multi-comments better, then only single-comments.

I'm don't talking about need right now create multi-comments feature. But I also believe this changes not give so much, than was before (+ break possible future multi-comments feature).

Anyway. it's not seems good, you know:
![image](https://user-images.githubusercontent.com/16944225/54682601-abc64080-4b0f-11e9-9c90-97bb1888a7e7.png)

Reply:
Emm, I see... https://github.com/iissnan/hexo-theme-next/issues/1549 and https://github.com/iissnan/hexo-theme-next/issues/1604

Reply:
Maybe same refactoring can be with multi-comments choice, in possible future, of course:

```diff
comments:
  # Type list:
  # Disqus | Disqusjs | Changyan | Valine | LiveRe | Gitment | Gitalk | facebook_comments_plugin | vkontakte

  # or multi-list:
  #Disqus
  #Valine
  #Gitalk
  ...
```
And for each function...

Reply:
@ivan-nginx like this <https://hexo.io/docs/deployment.html>, I think it's probably easier to implement a multi-comment system in this way in the future.

Reply:
Yes, like this.

Reply:
> Anyway, it's not seems good, you know (+ Breaking Changes):
> ![image](https://user-images.githubusercontent.com/16944225/54682601-abc64080-4b0f-11e9-9c90-97bb1888a7e7.png)

Sorry, this PR will inevitably affect almost all users. Due to comment plug-ins are a necessity.
But in fact, there hasn't been so much change, mostly migrating code in `post.swig` and `comments.swig` to separate files.